koku/api/provider/models.py

```diff
@@ -0,0 +1,71 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+"""Models for provider management."""
+
+from uuid import uuid4
+
+from django.db import models
+
+
+class ProviderAuthentication(models.Model):
+    """A Koku Provider Authentication.
+
+    Used for accessing cost providers data like AWS Accounts.
+    """
+
+    uuid = models.UUIDField(default=uuid4, editable=False,
+                            unique=True, null=False)
+    provider_resource_name = models.TextField(null=False)
+
+
+class ProviderBillingSource(models.Model):
```

Comment:
Cloud Health doesn't seem to need it:
http://apidocs.cloudhealthtech.com/#account_enable-aws-account

I think they are pulling the region from the ARN:
https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html#genref-arns

```
arn:partition:service:region:account-id:resource
arn:partition:service:region:account-id:resourcetype/resource
arn:partition:service:region:account-id:resourcetype:resource
```

Does that work for you?

Reply:
Or are we saying that since we are going to try and use Role-based cross-account access - https://docs.aws.amazon.com/IAM/latest/UserGuide/tutorial_cross-account-with-roles.html

and these have an ARN without a region typically based on the doc:
```
{
  "Version": "2012-10-17",
  "Statement": {
    "Effect": "Allow",
    "Action": "sts:AssumeRole",
    "Resource": "arn:aws:iam::PRODUCTION-ACCOUNT-ID:role/UpdateApp"
  }
}
```
We need this info to be explicit.